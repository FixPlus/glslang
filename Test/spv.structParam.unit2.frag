#version 460 core

struct MyStruct {
    vec3 v;
    int i;
};

int foo(inout MyStruct);

out float cout;

void main()
{
    MyStruct s;
    s.i = 2;
    int r = foo(s);
}
