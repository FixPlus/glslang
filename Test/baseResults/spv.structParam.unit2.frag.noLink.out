spv.structParam.unit2.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:12  Function Definition: main( ( global void)
0:12    Function Parameters: 
0:?     Sequence
0:15      move second child to first child ( temp highp int)
0:15        i: direct index for structure ( global highp int)
0:15          's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:15          Constant:
0:15            1 (const int)
0:15        Constant:
0:15          2 (const int)
0:16      Sequence
0:16        move second child to first child ( temp highp int)
0:16          'r' ( temp highp int)
0:16          Function Call: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:16            's' ( temp structure{ global highp 3-component vector of float v,  global highp int i})
0:?   Linker Objects
0:?     Linker Function: foo(struct-MyStruct-vf3-i11; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( inout structure{ global highp 3-component vector of float v,  global highp int i})
0:?     'cout' ( out highp float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 26

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 25
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 460
                              Name 4  "main"
                              Name 9  "MyStruct"
                              MemberName 9(MyStruct) 0  "v"
                              MemberName 9(MyStruct) 1  "i"
                              Name 13  "foo(struct-MyStruct-vf3-i11;"
                              Name 14  "s"
                              Name 19  "r"
                              Name 20  "param"
                              Name 25  "cout"
                              Decorate 13(foo(struct-MyStruct-vf3-i11;) Linkage Attributes 678391654 1970435187 1294824547 1920226169 762602357 758343286 993079657 0 1
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypeVector 6(float) 3
               8:             TypeInt 32 1
     9(MyStruct):             TypeStruct 7(fvec3) 8(int)
              10:             TypePointer Function 9(MyStruct)
              11:             TypeFunction 8(int) 10(ptr)
              15:      8(int) Constant 1
              16:      8(int) Constant 2
              17:             TypePointer Function 8(int)
              24:             TypePointer Output 6(float)
        25(cout):     24(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
           14(s):     10(ptr) Variable Function
           19(r):     17(ptr) Variable Function
       20(param):     10(ptr) Variable Function
              18:     17(ptr) AccessChain 14(s) 15
                              Store 18 16
              21: 9(MyStruct) Load 14(s)
                              Store 20(param) 21
              22:      8(int) FunctionCall 13(foo(struct-MyStruct-vf3-i11;) 20(param)
              23: 9(MyStruct) Load 20(param)
                              Store 14(s) 23
                              Store 19(r) 22
                              Return
                              FunctionEnd
13(foo(struct-MyStruct-vf3-i11;):      8(int) Function None 11
              12:     10(ptr) FunctionParameter
                              FunctionEnd
